sasmodels/kernelcuda.py

```diff
@@ -284,14 +284,15 @@ def compile_program(self, name, source, dtype, fast, timestamp):
         tag = generate.tag_source(source)
         key = "%s-%s-%s%s"%(name, dtype, tag, ("-fast" if fast else ""))
         # Check timestamp on program.
-        program, program_timestamp = self.compiled.get(key, (None, np.inf))
-        if program_timestamp < timestamp:
+        program, compile_timestamp, kernels = self.compiled.get(key, (None, np.inf, []))
+        if compile_timestamp < timestamp:
             del self.compiled[key]
         if key not in self.compiled:
             logging.info("building %s for CUDA", key)
             program = compile_model(str(source), dtype, fast)
-            self.compiled[key] = (program, timestamp)
-        return program
+            kernels = [getattr(program, k) for k in kernel_names]
+            self.compiled[key] = (program, timestamp, kernels)
```

Comment:
This should be failing with `TypeError: unhashable type: 'list'`, and similarly in kernelcl.